numpy/linalg/linalg.py

```diff
@@ -2141,7 +2141,10 @@ def lstsq(a, b, rcond="warn"):
 
     signature = 'DDd->Ddid' if isComplexType(t) else 'ddd->ddid'
     extobj = get_linalg_error_extobj(_raise_linalgerror_lstsq)
-    x, resids, rank, s = gufunc(a, b, rcond, signature=signature, extobj=extobj)
+    _b = zeros((m, n_rhs + 1), dtype=b.dtype) if is_empty_b else b
```

Comment:
I'd rather not take a step backwards on #10938 - this needs to be at minimum `b.shape[:-2] + (m, n_rhs + 1)`. It'd be nice if this could end up in the C code, but I found that was pretty awkward to do.

Reply:
What do you mean?

Reply:
Sorry, should have linked to #8720.

I've tried to set this function up so that in future it can work on stacks of arrays. Right now everything is in place except the stupid shape convention used by `resids` that doesn't make any sense

Reply:
With `matrix` being deprecated why would that be a problem?

But there is a need to be clear on what we mean by broadcasting...
```
a (-xyz-, m, n), b (-xyz-, m, n_rhs)
  |
 \|/
x (-xyz-, n, n_rhs), resids (-xyz-, n, n_rhs), rank (-xyz-), s (-xyz-, min(m,n))
```

Reply:
With broadcasting, the basic approach is ok with one argument, but with an `a` and a `b`,  naively wrapping the function would lead to unnecessary `svd`'s being run. But I guess composing the operations would make sense to reduce work. As in:

`lstsq2(svd(a), b)` wherein the outputs of `svd(a)` are passed into a function that supports broadcasting.

Reply:
An example of where what you have here would fail is `a.shape == (1, 5, m, n)`, `b.shape == (10, 1, m, n_rhs)`. The resulting components need to all have shape `(10, 5, ...)`, but your change would make them have shape `(1, 5, ...)`

Of course this is all sort of hypothetical anyway, because of the `_assertRank2` at the top of the function. I'd like to remove it eventually.

Reply:
Those internals are hidden away within the lapack functions, so they do not ensure that.

Reply:
> I see. Leave it be then.

I'd still prefer you to change it - it saves work down the line, and it makes the `[-2:]` slice above make more sense